### PR TITLE
Product details page fixed (Bugfix/492)

### PR DESCRIPTION
**Pull request by MMashivskij**
Code changes in the files "Product-details", "Comments" and "Similar-products".
All files in folder src/pages/product-details.

### CODE REVIEW

File: src/pages/product-details/comments/comments.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ const Comments = () => {
   const styles = useStyles();
   const dispatch = useDispatch();
 
-  const {
+  let {
```
Comment (AndriyLymych) [inline src/pages/product-details/comments/comments.js:33]: use const

File: src/pages/product-details/product-details.js [hunk 2]
```diff
@@ -46,23 +46,24 @@ const ProductDetails = ({ match }) => {
...
-  const { volumeInLiters, weightInKg } =
-    (defaultSize && defaultSize.size) || (options[0] && options[0].size) || {};
+  // const defaultSize = useMemo(
+  //   () =>
+  //   sizes[0]
+  //       ? sizes.find(( size ) => {
+  //         return size.name === DEFAULT_SIZE})
+  //       : {},
+  //   [sizes]
+  // );
```
Comment (AndriyLymych) [inline src/pages/product-details/product-details.js:64]: remove commented code

File: src/pages/product-details/product-details.js [hunk 3]
```diff
@@ -105,16 +106,16 @@ const ProductDetails = ({ match }) => {
...
+  // const uniqueSizes = useMemo(
+  //   () => [
+  //     ...new Set(
+  //       options.length && options[0].size
+  //         ? options.map(({ size: { available, name } }) => available && name)
+  //         : null
+  //     )
+  //   ],
+  //   [options]
+  // );
```
Comment (AndriyLymych) [inline src/pages/product-details/product-details.js:118]: same

File: src/pages/product-details/product-details.js [hunk 4]
```diff
@@ -156,19 +157,19 @@ const ProductDetails = ({ match }) => {
...
-  );
+  // const sizes = useMemo(
+  //   () =>
+  //     options.length
+  //       ? uniqueSizes.map(
+  //         (item) => options.find(({ size: { name } }) => item === name).size
+  //       )
+  //       : null,
+  //   [uniqueSizes, options]
+  // );
```
Comment (AndriyLymych) [inline src/pages/product-details/product-details.js:168]: same

File: src/pages/product-details/similar-products/similar-products.js [hunk 5]
```diff
@@ -12,16 +12,21 @@ import SimilarProductsItem from './similar-products-item';
...
+
   const { title } = SIMILAR_ITEMS[language];
 
+  if (similarProducts === null) {
+    similarProducts = [];
+  }
+
   const imagesList = similarProducts
     .filter(({ _id }) => _id !== productId)
     .map(({ _id, images, rate, name, basePrice }) => (
```
Comment (AndriyLymych) [inline src/pages/product-details/similar-products/similar-products.js:32]: use func from lodash for this operations

**Comment by AndriyLymych** [comment]
image
fix conflicts

**Comment by AndriyLymych** [comment]
image
fill this info

File: src/pages/product-details/comments/comments.js (hunk 1 above)
Comment (MMashivskij) [reply src/pages/product-details/comments/comments.js:33]: > use const

The value of the variables is reassigned. That`s why ussing "let".

**Comment by Kiyooooko** [comment]
link your PR to your issue and remove from board
Screenshot 2021-02-23 at 19 13 37

**Pull request closed without merging**